### PR TITLE
add option to choose names for Lanterns in Inventory and SWADE support

### PR DESCRIPTION
**Pull request by marcosmachado81**
- I added swade in the system check (it's work without needed for changes for actors items check)
- I created a function to check a list of items to accept add more items name as Lantern. Now, instead of use only 2 items name as lantern the GM can choose the items considered a lantern. the function hasLanternItemInInventory() check the item from settings "nameLanternList". You need to set items divided by ";". the default value is "Lantern, Hooded;Lantern, Bullseye"
- I removed "Oil (flask)" string from hardcode check and now I use "nameConsumableLantern" calling the function  game.settings.get("torchlight", "nameConsumableLantern")

### CODE REVIEW

**Comment by marcosmachado81** [comment]
> Thanks for all that, as you might have seen, I'm a bit of a noob in general and on gitbub in particular, so I'm not sure how to manage that but I'm sure to try. If you have any advice, I'm of course grateful.

I recommend you read [this link about checking pull requests](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally) before accepting changes. You will test the changes on the local repository and decide If you need to change something before the merge, or if you don't accept the changes. 

To test my changes you need to open the configuration, go to the torchlight section and change the torchlight.nameLantern.name (you can add many you want, just add a ";" between the names) and torchlight.nameConsumableLantern.name to change the consumable name that you need to have in inventory :)

**Comment by SMJL88** [comment]
Hey just wanted to comment on this. I made a similar change myself, although mine was just to fix the existing options to pull from the stored data, since the names were hard coded. marcosmachado81, just a thought but you may want to add the same functionality to the torch, as currently it specifically looks for "torch" and won't allow you to change the name of it.

I'm still new to github myself so it is possibly I'm misreading the suggested fixes.

**Comment by marcosmachado81** [comment]
Hello @SMJL88  could you check my last changes? https://github.com/marcosmachado81/TorchLight

maybe we can merge our work

File: torchlight.js [hunk 1]
```diff
@@ -1,1047 +1,1072 @@
...
+							app.object.getFlag("torchlight", "InitialLightAngle"),
+							app.object.getFlag("torchlight", "InitialAnimationType"),
+							app.object.getFlag("torchlight", "InitialAnimationSpeed"),
+							app.object.getFlag("torchlight", "InitialAnimationIntensity"),
+						);
+					} else {
+						// The token does not have the torch on
+						console.log("Clicked on the torch when the torch is off.");
+						// Checks whether the character can consume a torch
+						if (consumeItem("Torch")) {
```
Comment (SMJL88) [inline torchlight.js:244]: replace this line with 

if (consumeItem(game.settings.get("torchlight", "nameConsumableTorch"))) {

Comment (marcosmachado81) [reply torchlight.js:244]: Yes, I missed that.. I already did on my master branch. I am doing some changes to show messages when turning on or off a light/torch/lantern. but all changes are In my repo. I will close this pull request and will do a new one when I finish my changes.

**Pull request closed without merging**